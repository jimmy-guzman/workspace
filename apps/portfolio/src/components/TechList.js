import React from 'react'
import { arrayOf, string } from 'prop-types'

const TechList = ({ techs }) => (
  <ul style={{ textAlign: 'center' }}>
    {techs.map(tech => (
      <li className='badge' key={tech}>
        {tech}
      </li>
    ))}
  </ul>
)

TechList.propTypes = {
  techs: arrayOf(string).isRequired
}

export default TechList
