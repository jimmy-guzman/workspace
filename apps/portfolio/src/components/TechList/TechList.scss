@function em($target, $context: 18px) {
  @return ($target/$context) * 1em;
}

.TechList {
  font-family: 'Oswald', Impact, 'Franklin Gothic Bold', sans-serif;
}

.badge {
  display: inline-block;
  background: #d7d5d9;
  border-radius: em(3px);
  color: #292b2f;
  padding: em(5px);
  margin: em(6px);
}
