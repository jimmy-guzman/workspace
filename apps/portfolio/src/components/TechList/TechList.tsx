import React from 'react'

export interface TechListProps {
  technologies: string[]
}

export const TechList = ({ technologies }: TechListProps) => (
  <ul style={{ textAlign: 'center' }}>
    {technologies.map(technology => (
      <li className='badge' key={technology}>
        {technology}
      </li>
    ))}
  </ul>
)
