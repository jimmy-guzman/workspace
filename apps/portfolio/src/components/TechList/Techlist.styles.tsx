import styled from 'styled-components'

export const List = styled.ul`
  font-family: ${({ theme }) => theme.font.family};
  display: flex;
  flex-flow: wrap;
  justify-content: center;
`

export const Styled = { List }
