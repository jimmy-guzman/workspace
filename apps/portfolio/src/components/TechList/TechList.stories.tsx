import React from 'react'
import { TechList } from './TechList'

export default {
  title: 'portfolio/components/TechList',
  component: TechList,
}

export const normal = () => (
  <TechList technologies={['JavaScript', 'ReactJS', 'Redux']} />
)
