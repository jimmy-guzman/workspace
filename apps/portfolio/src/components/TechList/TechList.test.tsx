import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { render } from '@testing-library/react'

import { TechList, TechListProps } from './TechList'

const setupTechList = (
  props: TechListProps = {
    technologies: ['JavaScript', 'ReactJS', 'Redux'],
  }
) => {
  const history = createMemoryHistory()

  return render(
    <Router history={history}>
      <TechList {...props} />
    </Router>
  )
}

describe('<TechList />', () => {
  it('should render default snapshot', () => {
    const { container } = setupTechList()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render snapshot with not isEnglish', () => {
    const { container } = setupTechList()

    expect(container.firstChild).toMatchSnapshot()
  })
})
