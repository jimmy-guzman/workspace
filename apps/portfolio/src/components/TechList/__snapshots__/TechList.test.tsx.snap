// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechList /> should render default snapshot 1`] = `
.c2 {
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  display: inline-block;
  background: #d7d5d9;
  border-radius: 0.16666666666666666em;
  color: #292b2f;
  padding: 0.2777777777777778em;
}

.c0 {
  list-style: none;
  padding: 0;
  color: #292b2f;
  text-align: center;
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 0.3333333333333333em;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      JavaScript
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      ReactJS
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      Redux
    </span>
  </li>
</ul>
`;

exports[`<TechList /> should render snapshot with not isEnglish 1`] = `
.c2 {
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  display: inline-block;
  background: #d7d5d9;
  border-radius: 0.16666666666666666em;
  color: #292b2f;
  padding: 0.2777777777777778em;
}

.c0 {
  list-style: none;
  padding: 0;
  color: #292b2f;
  text-align: center;
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 0.3333333333333333em;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      JavaScript
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      ReactJS
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      Redux
    </span>
  </li>
</ul>
`;
